src/algorithms/Algorithm.cpp

```diff
@@ -37,9 +32,7 @@ State Algorithm<Hierarchy, Hypers, Mixing>::get_state_as_proto(
 
 //! \param un_val Unique value in Protobuf-object form
 //! \return       Matrix version of un_val
-template <template <class> class Hierarchy, class Hypers, class Mixing>
-Eigen::MatrixXd Algorithm<Hierarchy, Hypers, Mixing>::proto_param_to_matrix(
-    const Param &un_val) const {
+Eigen::MatrixXd Algorithm::proto_param_to_matrix(const Param &un_val) const {
```

Comment:
Sure, will do

Reply:
This function, as well as the other ones like write_clustering_to_file(), are specific to the appropriate proto classes. That is, there's no generic to_eigen() function. Should we move them out anyway? Is it just a matter of being declared in a proto_utils file instead of the class header?

Reply:
ok, keep it here for the moment, but I think we should have a better design of the Protos we're using (e.g. having a EigenVector and EigenMatrix proto), once we sort that out, we move all the proto-related stuff outside this class